docs/source/app_developers_guide/docker.rst

```diff
@@ -0,0 +1,417 @@
+
+This diagram represents a simple network with just two validators and two
+transaction processors. The second validator's transaction processors are not
+depicted.
+
+
+Installing Docker Engine and Docker Compose
+===========================================
+
+On Windows and macOS, install the latest version of `Docker Engine <https://docs.docker.com/engine/installation/>`_.
+
+  - On Windows and macOS, Docker Compose is installed automatically when you
+  - install Docker Engine
+
+On Linux, install both `Docker Engine <https://docs.docker.com/engine/installation/>`_ and `Docker Compose <https://docs.docker.com/compose/install/>`_.
+
+.. warning::
+
+  Note that the minimum version of Docker Engine necessary is 17.03.0-ce.
+  Linux distributions often ship with older versions of Docker.
+
+
+Installation
+============
+
+Download the Docker Compose File
+--------------------------------
+
+A Docker Compose file is provided which defines the process for constructing
+a Sawtooth environment, including the container images to download and the
+network settings.
+
+This docker compose file can serve as the basis for your own multi-container
+sawtooth development environment or application.
+
+Download the docker compose file `here <./sawtooth-default.yaml>`_.
+
+Environment Startup
+-------------------
+
+To start up the environment, perform the following tasks:
+
+1. Open a terminal window.
+2. Change your working directory to the same directory where you saved the`
+3. Run the following command:
+
+.. code-block:: console
+
+  % docker-compose -f sawtooth-default.yaml up
+
+Downloading the docker images that comprise the Sawtooth demo
+environment can take serveral minutes. Once you see the containers
+registering and creating intial blocks you can move on to the next step.
+
+.. code-block:: console
+
+  Attaching to compose_validator_1, compose_tp_xo_python_1, compose_client_1, compose_tp_intkey_python_1, compose_tp_config_1, compose_rest_api_1
+  validator_1         | writing file: /etc/sawtooth/keys/validator.priv
+  validator_1         | writing file: /etc/sawtooth/keys/validator.pub
+  validator_1         | Generating /var/lib/sawtooth/genesis.batch
+  tp_xo_python_1      | [19:03:47 DEBUG   selector_events] Using selector: ZMQSelector
+  validator_1         | [19:03:47.537 INFO     path] Skipping path loading from non-existent config file: /etc/sawtooth/path.toml
+  tp_xo_python_1      | [19:03:47 INFO    core] register attempt: OK
+  validator_1         | [19:03:47.538 INFO     cli] config [path]: config_dir = "/etc/sawtooth"
+  tp_intkey_python_1  | [19:03:47 DEBUG   selector_events] Using selector: ZMQSelector
+  validator_1         | [19:03:47.538 INFO     cli] config [path]: key_dir = "/etc/sawtooth/keys"
+  tp_intkey_python_1  | [19:03:47 INFO    core] register attempt: OK
+
+Open a new terminal so we can connect to the client container:
+
+.. code-block:: console
+
+  % docker exec -it compose_client_1 bash
```

Comment:
Right now, the container names from docker-compose are based on the folder where the compose file is located and thus aren't predictable. I submitted a PR to change this behavior:

https://github.com/hyperledger/sawtooth-core/pull/572